### ISSUE TITLE
validation-bug - parser confuses <thead> for <tbody>

### ISSUE THREAD

**Issue by kaizhu256** [validator/validator#1062]
see screenshot

![image](https://user-images.githubusercontent.com/280571/103038555-5fa1fe80-4534-11eb-8d0e-58a5bdc825fa.png)

html code:
```html
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>
<p></p>
<table>
    <thead>
        <th></th>
    </thead>
</table>
</body>
</html>
```

**Comment by sideshowbarker** [comment]
Agreed that error message is confusing.

I’m inclined to think it should just be changed to “`th` start tag not within `tr` element”.

But to be clear one thing: the “in table body” in that error message does not mean “in the `tbody` element”.

If you go to https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-intable and then scroll to the part which says, *“A start tag whose tag name is one of: "tbody", "tfoot", "thead"”*, you’ll notice it then says, *“switch the insertion mode to "in table body"”*.

That is, it says the parser must switch into the "in table body" insertion mode even though it actually could be within a `thead` or `tfoot` element — not just within a `tbody` element.

And then if you go to https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-intbody and see the part that says, *“A start tag whose tag name is one of: "th", "td"”*, that’s the part of the spec that this error message corresponds to.

**Comment by kaizhu256** [comment]
oh thx.  fixed the lint error now that i know what it is.  will leave issue open if there's interest in fixing the message.  otherwise feel free to close this.
